lib/streamlit/elements/layouts.py

```diff
@@ -29,11 +31,18 @@
 
 SpecType: TypeAlias = Union[int, Sequence[Union[int, float]]]
 
+# Pattern to validate a container key (to be compatible with css class names):
+_KEY_PATTERN: Final = re.compile(r"^[a-zA-Z_][a-zA-Z0-9\-_]*$")
```

Comment:
The check here that it starts with a letter is not needed since we prefix the key anyways with some letters (e.g. `st-`)